### DISCUSSION TITLE
Catchall alias not working

### DISCUSSION THREAD

**Discussion by reesing322** [modoboa/modoboa#2194] [Q&A] [answered]
I have added a new account, like catchall@test.net, and also gave it an alias @test.net ,
I thought that was the right way to add a catchall for test.net.

But now, when I test it by sending an email to bla@test.net, the sender gets an instant failure response,
and the syslog shows something like
said: 550 5.1.1 <bla@test.net> User doesn't exist: bla@test.net (in reply to RCPT TO command);

Do I still need to tweak postfix config  somehow?

**Comment by reesing322** [comment]
I just managed to get the catchall working by adding a proxy to virtual_alias_maps in main.cf, and that proxy executes this call:
query = SELECT alr.address FROM modoboa_admin_aliasrecipient AS alr INNER JOIN admin_alias AS al ON alr.alias_id=al.id WHERE al.enabled AND al.address=(select right('%s', 1+char_length('%s')-strpos('%s', '@'))) AND (al.expire_at IS NULL OR al.expire_at>now())

Somehow I hope there is a simpler solution?

**Comment by tonioo** [comment, marked as answer]
It might a verify cache issue. Try to delete your cache and retry.

**Reply by reesing322** [reply]
Ok, ... how/where?

**Reply by reesing322** [reply]
Found it,  /var/lib/postfix/verify_cache.db
Indeed, stopping postfix, removing that file and restarting postfix seems to solve the problem.
Any idea why the cache was in an incorrect state? Does this mean the cache can't be trusted?

**Reply by tonioo** [reply]
The cache has a default expiry time. If you try to send a message to an address which does not exist, then create the address and retry, you'll have to wait until the cache expiry... I think you can tune this, look at postfix's doc.
